### ISSUE TITLE
Modelica.Blocks.Continuous.LimIntegrator limitsAtInit implmentation

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#751]
**Reported by garron&#x2e;fish on 28 May 2012 14:55 UTC**
In the Modelica.Blocks.Continuous.LimIntegrator component:

The implmentation of limitsAtInit is not implmented as per documentation.  

When limitAtInit is true then the limits are ignored at initialisation if limitAtInit is false then the limits are used.  The documentation states:
If parameter limitAtInit = false, the limits of the integrator are removed ... 

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/751

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 29 May 2012 05:49 UTC**
I looked at the block and what I can gather from the code is that it is programmed to behave like the documentation states:

```mo
equation 
  if initial() and not limitsAtInit then
     der(y) = k*u;
 ...
  else
     der(y) = if y < outMin and u < 0 or y > outMax and u > 0 then 0 else k*u;
  end if;
```
So I do not see a problem here.

**Comment by modelica-trac-importer** [comment]
**Comment by garron&#46;fish on 29 May 2012 08:42 UTC**
I misunderstood the implemntation of the model, so the reason for this ticket is invalid.

I was confused by continuing a simulation that contained 0 as a limit.  Numerical errors resulted in the assert statement being invalid. Altering the assert statement to something like the following may be an improvement:

```
    assert(y >= outMin - 0.0001*(outMax-outMin) and y <= outMax + 0.0001*(outMax-outMin), "LimIntegrator: During initialization the limits have been ignored.\n"
       + "However, the result is that the output y is not within the required limits:\n"
       + "  y = " + String(y) + ", outMin = " + String(outMin) + ", outMax = " + 
      String(outMax));
```

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 29 May 2012 08:47 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 2 Jan 2013 22:37 UTC**

**Comment by modelica-trac-importer** [comment]
**Changelog removed by otter on 2 Jan 2013 22:37 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 5 Jan 2013 19:10 UTC**
Fixed in 758bdc137cf4a77e6ea57923ebfa8882b922af09:

As proposed above but with relative tolerance of 0.001 instead of 0.0001
(changed assert, so that values of outMin=0 or outMax=0, still yield an assertion where the real limits are a bit outside of the required limits to cope with numerical imprecision).
